pancancer_evaluation/utilities/file_utilities.py

```diff
@@ -9,6 +9,50 @@
 
 from pancancer_evaluation.exceptions import ResultsFileExistsError
 
+def construct_filename(output_dir,
+                       file_descriptor,
+                       extension,
+                       *args,
+                       **kwargs):
+    """Construct a filename from varying experimental parameters.
+
+    The format of the filename will roughly look like this:
+    {output_dir}/{a}_{k}_{file_descriptor}{extension}
+    where a = '_'.join([ar for ar in args]),
+          k = '_'.join(['{k}{v} for k, v in kwargs.items()])
+
+    For example:
+    >>> construct_filename('output_dir', 'output', '.tsv',
```

Comment:
Is having the ">>>" and "..." for code written inside a docstring standard? It's kind of throwing me off

Reply:
It is - you can actually run them directly as unit tests with the [doctest package](https://docs.python.org/3.4/library/doctest.html) (which is a built-in part of the Python standard library).